### ISSUE TITLE
Category Income/Expenses Report (Monthly) - enhancement

### ISSUE THREAD

**Issue by junkie765** [moneymanagerex/moneymanagerex#4039]
# Description of the feature

Currently the report is showing all data in one table. It's hard understand which categories are income categories and which are expense categories.
Requested functionality:
1. Have two separate sections  

- one for the income categories and the respective subtotals
- second for expense categories and the respective subtotals
- Totals at the bottom
- if the sub-categories can be expanded/collapsed, that would really nice
2. Have period selection as other reports - current year, last year, last month, custom range, etc.

I'm thinking something like this:
<img width="466" alt="image" src="https://user-images.githubusercontent.com/54117504/144434801-0cff1f39-6d18-4a18-a617-a4dde8f48bcb.png">

**Comment by whalley** [comment]
We do not distinguish between income and expense categories. You can have income and expenses against a single category. 

I agree that having a a customisable range may be beneficial but would need to consider how many months we can sensible display horizontally.

**Comment by junkie765** [comment]
> We do not distinguish between income and expense categories. You can have income and expenses against a single category.
> 
> I agree that having a a customisable range may be beneficial but would need to consider how many months we can sensible display horizontally.

Maybe it's nice to have income and expense categories. As example - incomes - salary, dividends, etc. Everything else can be expense categories. The user should be able to define. And of course there should be no restriction of what kind of transactions are added to a category - it can be both debit and credit in an expense category.
As for the date ranges, it can be any period selectable by the user. The same way you have it for other reports. This is what I mean specifically.
<img width="332" alt="image" src="https://user-images.githubusercontent.com/54117504/144744148-3aa2259f-02fd-4ca8-9e60-5648aba516b1.png">

**Comment by whalley** [comment]
**QUESTION** - As we don't currently have categories classified as INCOME or EXPENSE we cannot create this report. What do people feel about adding a classification to the categories? Subordinate categories would have the same classification as there parent category.

Personally I have no use for this, but it seems others do.

**Comment by tactilis** [comment]
> What do people feel about adding a classification to the categories?

Issues to consider:

1. It is legitimate to use the same category for Deposit and Withdrawal transactions (despite what the Note in the Help says).

An example of this is expenditure on an particular item (a withdrawal/expense for a particular category), for which you later receive a full or partial refund (i.e. deposit/income). The way it is suggested that this be handled is to give the deposit the _same_ category and mark the transaction as a Refund (either in the Notes or the Number field). This would not work if the categories were defined as INCOME or EXPENSE.

2. Some categories, such as Transfer, are neither INCOME, nor EXPENSE.

3. How would you apply the INCOME and EXPENSE classification to existing categories. Would the user have to go though them and specify INCOME/EXPENSE for each.

**Comment by whalley** [comment]
@tactilis You raise good points. 

I really don't want income and expense categories so would not want use either of those classifications - so another classification ("INCOME AND EXPENSE") which is what we have now and what we use prior to anyone classifying otherwise. E.g I often do shopping for my elderly mother so just have a category for that, it's neither a dedicated Expense or Income. It just a category that balances out when she pays me.

A TRANSFER could be restricted to "INCOME AND EXPENSE" (is there a better name?) categories

It would need quite a bit of thinking in how we present this in the reporting.

**Comment by tactilis** [comment]
@whalley
> so another classification ("INCOME AND EXPENSE") which is what we have now and what we use prior to anyone classifying otherwise

> A TRANSFER could be restricted to "INCOME AND EXPENSE" (is there a better name?) categories

> It would need quite a bit of thinking in how we present this in the reporting.

Yes. It appears that we are layering on more and more complexity if we go down this route.

So, now that we have multi-level categories and easy reorganisation of the category tree, would a workaround option for those who want this sort of INCOME/EXPENSE classification be for _them_ to expand the classification hierarchy with an additional 3 top level categories. For example:

```
E:
    Bills:
        Electricity
        Gas
        etc
    Car:
    etc
I:
    Investments
    Salary
    etc
X:
    Transfer
    etc
```

Where:
E = Expense
I = Income
X = Neither

Would that fit more easily into the current method of reporting?

**Comment by whalley** [comment]
@tactilis I think your suggestion would help in terms of presentation on some reports and overall classification of categories. However the term 'Expense' / 'Income' on the category reports would of course continue to represent a -ve aggregated value for the category over the period / -ve aggregated value for the category over the period respectfully. Which should be OK in 99.9% of the cases.

@junkie765 Would this meet you needs for separating Expense / Income categories

**Comment by whalley** [comment]
Now we have nestable categories and user can create an Income and Expense catgeory and then create sub-categories under these does this enhancemnt request now just need the ability to specify a period range for the report?

**Comment by junkie765** [comment]
> @tactilis I think your suggestion would help in terms of presentation on some reports and overall classification of categories. However the term 'Expense' / 'Income' on the category reports would of course continue to represent a -ve aggregated value for the category over the period / -ve aggregated value for the category over the period respectfully. Which should be OK in 99.9% of the cases.
> 
> @junkie765 Would this meet you needs for separating Expense / Income categories

Yes,  this would be sufficient.
